theano/tensor/raw_random.py

```diff
@@ -520,13 +520,8 @@ def binomial(random_state, size=None, n=1, p=0.5, ndim=None,
     n = tensor.as_tensor_variable(n)
     p = tensor.as_tensor_variable(p)
     ndim, size, bcast = _infer_ndim_bcast(ndim, size, n, p)
-    if n.dtype == 'int64':
-        # THIS WORKS AROUND A NUMPY BUG on 32bit machine
-        # Erase when the following works on a 32bit machine:
-        # numpy.random.binomial(
-        #          n=numpy.asarray([2,3,4], dtype='int64'),
-        #          p=numpy.asarray([.1, .2, .3], dtype='float64'))
-        n = tensor.cast(n, 'int32')
+    # THIS WORKS AROUND A NUMPY BUG on 32bit machine
```

Comment:
Actually, there is an issue here. I checked the case in the comment, and there is still an issue on 32-bits installations of Python. So this change would probably break things...